### PR TITLE
Change internal_encode() to use FR_DBUFF_NO_ADVANCE()

### PR DESCRIPTION
**Pull request by jejones3141**

### CODE REVIEW

File: src/protocols/internal/encode.c [hunk 1]
```diff
@@ -57,38 +57,31 @@ static ssize_t internal_encode(fr_dbuff_t *dbuff,
...
 
 	FR_PROTO_STACK_PRINT(da_stack, depth);
 
-	enc_field = dbuff->p;
+	enc = work_dbuff.p;
 
 	/*
 	 *	Zero out first encoding byte
 	 */
-	fr_dbuff_memset(dbuff, 0, 1);
```
Comment (arr2036) [inline src/protocols/internal/encode.c:76]: Why was this removed?

File: src/protocols/internal/encode.c [hunk 2]
```diff
@@ -175,6 +172,7 @@ static ssize_t internal_encode(fr_dbuff_t *dbuff,
 	 */
 	case FR_TYPE_EXTENDED:	/* Just another type of TLV */
 	case FR_TYPE_TLV:
+	{
```
Comment (arr2036) [inline src/protocols/internal/encode.c:175]: Unnecessary bracketing?

File: src/protocols/internal/encode.c [hunk 3]
```diff
@@ -57,38 +57,31 @@ static ssize_t internal_encode(fr_dbuff_t *dbuff,
 			       fr_da_stack_t *da_stack, unsigned int depth,
 			       fr_cursor_t *cursor, void *encoder_ctx)
 {
-	uint8_t				*enc_field, *len_field, *value_field, *value_end;
```
Comment (arr2036) [inline src/protocols/internal/encode.c:60]: I really dislike fields being renamed for no purpose.

File: src/protocols/internal/encode.c (hunk 1 above)
Comment (jejones3141) [reply src/protocols/internal/encode.c:76]: Looking at the code, it's now `FR_DBUFF_BYTES_IN_RETURN(&work_dbuff, 0);`. If that's not there, please let me know.

File: src/protocols/internal/encode.c (hunk 2 above)
Comment (jejones3141) [reply src/protocols/internal/encode.c:175]: It is indeed unneccessary; out it goes.

File: src/protocols/internal/encode.c (hunk 3 above)
Comment (jejones3141) [reply src/protocols/internal/encode.c:60]: Quite understandable; back they go.

File: src/protocols/internal/encode.c (hunk 1 above)
Comment (arr2036) [reply src/protocols/internal/encode.c:76]: Ah, ok, apologies, missed that.

File: src/protocols/internal/encode.c (hunk 2 above)
Comment (jejones3141) [reply src/protocols/internal/encode.c]: Gone; the corrected version is pushed.

File: src/protocols/internal/encode.c (hunk 3 above)
Comment (jejones3141) [reply src/protocols/internal/encode.c:60]: Names restored and the result pushed.

**Comment by jejones3141** [comment]
Closing; a version that not only uses FR_DBUFF_NO_ADVANCE() but also follows other requested changes (avoiding fr_dbuff_remaining() and FR_DBUFF_CHECK_REMAINING_RETURN()) is in fr_struct_to_network_dbuff.
